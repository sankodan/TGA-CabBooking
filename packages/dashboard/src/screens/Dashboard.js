import React, { useEffect, useState } from 'react'
import { Background, Paragraph, Avatar, Button } from '@sankodan/com.cabbooking.components'
import { ActivityIndicator, FlatList, View } from 'react-native'
import apiManager from '@sankodan/com.cabbooking.core/src/api/axiosInstance'

export default function Dashboard (props) {
  const apiForMovies = 'https://api.npoint.io/f0263d463ab78b8b0c10'
  const [movies, setMovies] = useState([])
  /* istanbul ignore next  */
  useEffect(() => {
    apiManager
      .get(apiForMovies)
      .then((response) => {
        setMovies(response.data.movies)
      })
      .catch((error) => {
        console.error(error)
      })
  }, [])

  return (
    <Background>
      {movies && movies.length > 0
        ? <FlatList
            style={{ width: '100%' }}
            data={movies}
            keyExtractor={item => item.id}
            renderItem={({ item }) => {
              return (
                <View style={{ flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center' }}>
                  <Avatar url={item.image} />
                  <Paragraph style={{
                    lineHeight: 16
                  }}>{item.text}</Paragraph>
                  <View style={{ flex: 1 }}>
                    <Button testID="bookButton" mode="outlined" onPress={() => props.navigation.navigate('CheckOut', { item })}>
                      Book Now
                    </Button>
                  </View>
                </View>
              )
            }}
            numColumns={2}
            showsVerticalScrollIndicator={false}
        />
        : <ActivityIndicator /> }
    </Background>
  )
}
