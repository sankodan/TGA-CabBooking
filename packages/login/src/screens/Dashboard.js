import React from 'react'
import { Background, Logo, Header, Paragraph, Button } from '@sankodan/com.cabbooking.components'
import { logoutUser } from '@sankodan/com.cabbooking.core'

export default function Dashboard () {
  return (
    <Background>
      <Logo />
      <Header>Letâ€™s start</Header>
      <Paragraph>
        Your amazing app starts here. Open you favorite code editor and start
        editing this project.
      </Paragraph>
      <Button testID = 'logoutButton' mode="outlined" onPress={logoutUser}>
        Logout
      </Button>
    </Background>
  )
}
