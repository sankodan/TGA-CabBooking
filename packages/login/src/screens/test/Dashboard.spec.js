import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react-native'
import Dashboard from '../Dashboard'
import { shallow } from 'enzyme'

describe('Dashboard', () => {
  let toJSONObject
  describe('clicking logoutButton', () => {
    beforeEach(() => {
      const { toJSON } = render(<Dashboard/>)
      toJSONObject = toJSON
      fireEvent.press(screen.getByTestId('logoutButton'))
    })

    it('Dashboard snapshot', () => {
      expect(toJSONObject()).toMatchSnapshot()
    })
  })
})
