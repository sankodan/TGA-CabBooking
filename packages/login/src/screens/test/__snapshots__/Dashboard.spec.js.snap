// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard clicking logoutButton Dashboard snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../packages/components/assets/background_dot.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../packages/components/assets/logo.png",
        }
      }
      style={
        Object {
          "height": 110,
          "marginBottom": 8,
          "width": 110,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#121330",
            "fontSize": 21,
            "fontWeight": "bold",
            "paddingVertical": 12,
          },
        ]
      }
    >
      Letâ€™s start
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "fontSize": 15,
            "lineHeight": 21,
            "marginBottom": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      Your amazing app starts here. Open you favorite code editor and start editing this project.
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.29)",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "elevation": 0,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="logoutButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
